### PR TITLE
Bug 1987250: Remove diskEligible check for OCS

### PR DESCRIPTION
**Pull request by rewantsoni**
# Assisted Pull Request

## Description
OCS uses diskEligible field in the disk struct to check if the disk is eligible or not. As the OCP disk size is going to be increased to 30GB and OCS requires disks to be 25GB, validations will fail. This PR removes the dependency adds a check to ignore the disk if the disk size in 0

## List all the issues related to this PR

- [ ] New Feature
- [x] Bug fix
- [ ] Tests
- [ ] Documentation
- [ ] CI/CD 

## What environments does this code impact?

- [ ] Automation (CI, tools, etc)
- [ ] Cloud
- [ ] Operator Managed Deployments
- [x] None

## How was this code tested?

- [ ] assisted-test-infra environment
- [ ] dev-scripts environment
- [ ] Reviewer's test appreciated
- [ ] Waiting for CI to do a full test run
- [ ] Manual (Elaborate on how it was tested)
- [x] No tests needed

## Assignees

/cc @gobindadas 

## Checklist

- [x] Title and description added to both, commit and PR.
- [x] Relevant issues have been associated (see [CONTRIBUTING] guide)
- [x] Reviewers have been listed
- [x] This change does not require a documentation update (docstring, `docs`, README, etc)
- [x] Does this change include unit-tests (note that code changes require unit-tests)

## Reviewers Checklist

- [ ] Are the title and description (in both PR and commit) meaningful and clear?
- [ ] Is there a bug required (and linked) for this change?
- [ ] Should this PR be backported?

[Kubernetes community documentation]: https://github.com/kubernetes/community/blob/master/contributors/guide/pull-requests.md#commit-message-guidelines
[CONTRIBUTING]: https://github.com/openshift/assisted-service/blob/master/CONTRIBUTING.md

Closes: [OCPBUGSM-32645](https://issues.redhat.com/browse/OCPBUGSM-32645)

Signed-off-by: Rewant Soni <resoni@redhat.com>

### CODE REVIEW

**Review by gamli75 — APPROVED** [review]

File: internal/operators/ocs/validations.go [hunk 1]
```diff
@@ -168,7 +168,7 @@ func (o *operator) getValidDiskCount(disks []*models.Disk, installationDiskID st
 	var err error
 
 	for _, disk := range disks {
-		if (disk.DriveType == ssdDrive || disk.DriveType == hddDrive) && installationDiskID != disk.ID && disk.InstallationEligibility.Eligible {
+		if (disk.DriveType == ssdDrive || disk.DriveType == hddDrive) && installationDiskID != disk.ID && disk.SizeBytes != 0 {
```
Comment (gobindadas) [inline internal/operators/ocs/validations.go:171]: We also need to make sure disksize should not be lees than 25G

Comment (rewantsoni) [reply internal/operators/ocs/validations.go:171]: Yes, we are checking that in the next if statement, https://github.com/openshift/assisted-service/pull/2308/files#diff-271f123644f77e7b665d00eab988d90ab0275fde8eedcad9e36ea263592ffefbR172

Comment (gobindadas) [reply internal/operators/ocs/validations.go:171]: if (disk.DriveType == ssdDrive || disk.DriveType == hddDrive) && installationDiskID != disk.ID && disk.SizeBytes != 0)
when this condition does not meetthen what error we are throwing?

Comment (rewantsoni) [reply internal/operators/ocs/validations.go:171]: We are not throwing an error if this condition is not meeting, we are ignoring the disk as it's either not of the desired type, or its size is zero or it has already been marked for installation of OCP

Comment (gobindadas) [reply internal/operators/ocs/validations.go:171]: Great!

Comment (gobindadas) [reply internal/operators/ocs/validations.go:171]: /LGTM

**Review by gobindadas — APPROVED** [review]

**Pull request merged**